src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,30 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_SIZE_OF_PASSWORD = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() != null && user.getLogin() != "") {
+            User findUser = storageDao.get(user.getLogin());
+            if (findUser != null) {
+                throw new RuntimeException("The user with such login already exists. "
+                        + "Please, enter another login.");
+            }
+            if (user.getAge() < MIN_AGE) {
+                throw new RuntimeException("The user should be at least "
+                        + MIN_AGE + " years old.");
+            }
+            if (user.getPassword().length() < MIN_SIZE_OF_PASSWORD) {
+                throw new RuntimeException("Password should be at least "
+                        + MIN_SIZE_OF_PASSWORD + " characters.");
+            }
+            storageDao.add(user);
+            return user;
+        } else {
+            throw new NullPointerException("Login can't be null or empty!");
+        }
```

Comment:
It is bad practice to throw NPE, better to throw some RuntimeException. And lets implement this check like 
``` java
if (user.getLogin() == null || user.getLogin().isEmpty()) {
	throw new RuntimeException(....)
}
```